classes/User.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * User Class
+ *
+ * @package TheyWorkForYou
+ */
+
+namespace MySociety\TheyWorkForYou;
+
+/**
+ * User
+ */
+
+class User {
+    public function getMP($cons_type, $mp_house) {
+        $mp_data = array();
+        global $THEUSER;
+
+        if ($THEUSER->has_postcode()) {
+            // User is logged in and has a postcode, or not logged in with a cookied postcode.
+
+            // (We don't allow the user to search for a postcode if they
+            // already have one set in their prefs.)
+
+            // this is for people who have e.g. an English postcode looking at the
+            // Scottish homepage
+            try {
+                $constituencies = postcode_to_constituencies($THEUSER->postcode());
+                if ( isset($constituencies[$cons_type]) ) {
+                    $constituency = $constituencies[$cons_type];
+                    $MEMBER = new Member(array('constituency'=>$constituency, 'house'=> $mp_house));
+                }
+            } catch ( MemberException $e ) {
+                return $mp_data;
```

Comment:
Perhaps clearer to just return array() here, not define $mp_data above, and reduce the complexity of whole function by having a return array() shortcut if has_postcode is not true.